src/set_theory/ordinal_arithmetic.lean

```diff
@@ -1515,6 +1521,89 @@ theorem is_normal.eq_iff_zero_and_succ {f : ordinal.{u} → ordinal.{u}} (hf : i
   exact H b hb
 end)⟩
 
+/-! ### Minimum excluded ordinals -/
+
+/-- The minimum excluded ordinal in a family of ordinals. -/
+def mex {ι : Type u} (f : ι → ordinal.{max u v}) : ordinal :=
+Inf (set.range f)ᶜ
+
+theorem mex_nmem_range {ι : Type u} (f : ι → ordinal.{max u v}) : mex f ∉ set.range f :=
+Inf_mem (nonempty_compl_range f)
+
+theorem ne_mex {ι} (f : ι → ordinal) : ∀ i, f i ≠ mex f :=
+by simpa using mex_nmem_range f
+
+theorem mex_le_of_ne {ι} {f : ι → ordinal} {a} (ha : ∀ i, f i ≠ a) : mex f ≤ a :=
+cInf_le' (by simp [ha])
+
+theorem exists_of_lt_mex {ι} {f : ι → ordinal} {a} (ha : a < mex f) : ∃ i, f i = a :=
+by { by_contra' ha', exact ha.not_le (mex_le_of_ne ha') }
+
+theorem mex_le_lsub {ι} (f : ι → ordinal) : mex f ≤ lsub f :=
+cInf_le' (lsub_nmem_range f)
+
+theorem mex_monotone {α β} {f : α → ordinal} {g : β → ordinal} (h : set.range f ⊆ set.range g) :
+  mex f ≤ mex g :=
+begin
+  refine mex_le_of_ne (λ i hi, _),
+  cases h ⟨i, rfl⟩ with j hj,
+  rw ←hj at hi,
+  exact ne_mex g j hi
+end
+
+theorem mex_lt_ord_succ_mk {ι} (f : ι → ordinal) : mex f < (#ι).succ.ord :=
+begin
+  by_contra' h,
+  apply not_le_of_lt (cardinal.lt_succ_self (#ι)),
+  have H := λ a, exists_of_lt_mex ((typein_lt_self a).trans_le h),
+  let g : (#ι).succ.ord.out.α → ι := λ a, classical.some (H a),
+  have hg : function.injective g := λ a b h', begin
+    have Hf : ∀ x, f (g x) = typein (<) x := λ a, classical.some_spec (H a),
+    apply_fun f at h',
+    rwa [Hf, Hf, typein_inj] at h'
+  end,
+  convert cardinal.mk_le_of_injective hg,
+  rw cardinal.mk_ord_out
+end
+
+/-- The minimum excluded ordinal of a family of ordinals indexed by the set of ordinals less than
+    some `o : ordinal.{u}`. This is a special case of `mex` over the family provided by
+    `family_of_bfamily`.
+
+    This is to `mex` as `bsup` is to `sup`. -/
+def bmex (o : ordinal) (f : Π a < o, ordinal) : ordinal :=
+mex (family_of_bfamily o f)
+
+theorem bmex_nmem_brange {o : ordinal} (f : Π a < o, ordinal) : bmex o f ∉ brange o f :=
```

Comment:
```suggestion
theorem bmex_not_mem_brange {o : ordinal} (f : Π a < o, ordinal) : bmex o f ∉ brange o f :=
```

Reply:
I've renamed `lsub_nmem_range` to `lsub_not_mem_range` as well.